File: ExamplePlugin/src/main/java/me/andreasmelone/exampleplugin/Items.java
package me.andreasmelone.exampleplugin;

import me.andreasmelone.amutillib.i18n.TranslationKey;
import me.andreasmelone.amutillib.items.AMItem;
import me.andreasmelone.amutillib.items.ItemBuilder;
import me.andreasmelone.amutillib.items.ItemRegister;
import me.andreasmelone.amutillib.registry.RegisteredObject;
import me.andreasmelone.amutillib.utils.Util;
import org.bukkit.Material;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemFlag;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

public class Items {
    static ExamplePlugin plugin = ExamplePlugin.getInstance();
    public static RegisteredObject<AMItem> exampleItem = ItemRegister.getInstance().register(
            ItemBuilder.createBuilder(plugin, "example_item")
                    .setName("&rExample Item")
                    .setLore("&rThis is an example item.")
                    .setMaterial(Material.STONE_SHOVEL)
                    .build()
    );

    public static RegisteredObject<AMItem> shitItem = ItemRegister.getInstance().register(
            ItemBuilder.createBuilder(plugin, "shit")
                    .setName("&rShit")
                    .setLore("&r&fStinks.")
                    .setMaterial(Material.BROWN_DYE)
                    .build()
    );

    public static void register() {
        // EXAMPLE_ITEM START
        exampleItem.get().onCreateItemStack(event -> {
            event.getItemStack().addEnchantment(Enchantment.DIG_SPEED, 3);

            ItemMeta meta = event.getItemStack().getItemMeta();
            meta.setUnbreakable(true);
            meta.addItemFlags(ItemFlag.HIDE_ENCHANTS);
            event.getItemStack().setItemMeta(meta);
        });

        exampleItem.get().onInteract((event) -> {
            if(Util.isRightClick(event.getAction())) {
                if(event.getClickedBlock() != null)
                    event.getPlayer().sendMessage(plugin.getI18n().getTransformed(
                            "example_item.interact",
                            TranslationKey.of("%block%", Util.nameFromType(event.getClickedBlock().getType()))
                    ));
                else event.getPlayer().sendMessage(plugin.getI18n().getTransformed(
                        "example_item.interact_nothing"
                ));
            }
        });

        exampleItem.get().onBlockBreak((event) -> {
            event.getPlayer().sendMessage(plugin.getI18n().getTransformed(
                    "example_item.block_break",
                    TranslationKey.of("%block%", Util.nameFromType(event.getBlock().getType()))
            ));
        });

        exampleItem.get().onEntityHit((event) -> {
            event.getDamager().sendMessage(plugin.getI18n().getTransformed(
                    "example_item.entity_hit",
                    TranslationKey.of("%entity%", Util.nameFromType(event.getEntity().getType()))
            ));
        });
        // EXAMPLE_ITEM END

        // SHIT START
        shitItem.get().onInteract(event -> {
            Player player = event.getPlayer();

            int amount = event.getItem().getAmount();
            if(amount == 1) player.getInventory().setItem(event.getHand(), null);
            else event.getItem().setAmount(amount -1);

            player.sendMessage(plugin.getI18n().getTransformed("shit.eaten"));
            PotionEffect nausea = new PotionEffect(PotionEffectType.CONFUSION, 10 * 20, 1);
            player.addPotionEffect(nausea);
        });
        // SHIT END
    }
}

File: ExamplePluginKotlin/src/main/java/me/andreasmelone/examplepluginkotlin/Items.kt
package me.andreasmelone.examplepluginkotlin

import me.andreasmelone.amutillib.i18n.TranslationKey
import me.andreasmelone.amutillib.items.AMItem
import me.andreasmelone.amutillib.items.ItemBuilder
import me.andreasmelone.amutillib.items.ItemRegister
import me.andreasmelone.amutillib.items.events.CreateItemStackEvent
import me.andreasmelone.amutillib.registry.RegisteredObject
import me.andreasmelone.amutillib.utils.Util
import me.andreasmelone.exampleplugin.Items
import org.bukkit.Material
import org.bukkit.enchantments.Enchantment
import org.bukkit.event.block.BlockBreakEvent
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.inventory.ItemFlag
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType

class Items {
    private val plugin: ExamplePluginKotlin = ExamplePluginKotlin.getPlugin()

    companion object {
        private val plugin: ExamplePluginKotlin = ExamplePluginKotlin.getPlugin()
        val exampleItem: RegisteredObject<AMItem> = ItemRegister.getInstance().register(
                ItemBuilder.createBuilder(plugin, "example_item")
                        .setName("&rExample Item")
                        .setLore("&rThis is an example item.")
                        .setMaterial(Material.STONE_SHOVEL)
                        .build()
        )

        val shit: RegisteredObject<AMItem> = ItemRegister.getInstance().register(
                ItemBuilder.createBuilder(plugin, "shit")
                        .setName("&rShit")
                        .setLore("&r&fStinks.")
                        .setMaterial(Material.BROWN_DYE)
                        .build()
        )

        fun register() {
            // EXAMPLE_ITEM START
            Items.exampleItem.get().onCreateItemStack { event: CreateItemStackEvent ->
                event.itemStack.addEnchantment(Enchantment.DIG_SPEED, 3)

                val meta = event.itemStack.itemMeta
                meta!!.isUnbreakable = true
                meta!!.addItemFlags(ItemFlag.HIDE_ENCHANTS)
                event.itemStack.setItemMeta(meta)
            }

            Items.exampleItem.get().onInteract { event: PlayerInteractEvent ->
                if (Util.isRightClick(event.action)) {
                    if (event.clickedBlock != null) event.player.sendMessage(plugin.getI18n().getTransformed(
                            "example_item.interact",
                            TranslationKey.of("%block%", Util.nameFromType(event.clickedBlock!!.type))
                    ))
                    else event.player.sendMessage(plugin.getI18n().getTransformed(
                            "example_item.interact_nothing"
                    ))
                }
            }

            Items.exampleItem.get().onBlockBreak { event: BlockBreakEvent ->
                event.player.sendMessage(plugin.getI18n().getTransformed(
                        "example_item.block_break",
                        TranslationKey.of("%block%", Util.nameFromType(event.block.type))
                ))
            }

            Items.exampleItem.get().onEntityHit { event: EntityDamageByEntityEvent ->
                event.damager.sendMessage(plugin.getI18n().getTransformed(
                        "example_item.entity_hit",
                        TranslationKey.of("%entity%", Util.nameFromType(event.entity.type))
                ))
            }
            // EXAMPLE_ITEM END

            // SHIT START
            Items.shitItem.get().onInteract { event: PlayerInteractEvent ->
                val player = event.player
                val amount = event.item!!.amount
                if (amount == 1) player.inventory.setItem(event.hand!!, null)
                else event.item!!.amount = amount - 1

                player.sendMessage(plugin.getI18n().getTransformed("shit.eaten"))
                val nausea = PotionEffect(PotionEffectType.CONFUSION, 10 * 20, 1)
                player.addPotionEffect(nausea)
            }
            // SHIT END
        }
    }
}